src/main/scala/util/RecordListMap.scala

```diff
@@ -0,0 +1,26 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.util
+
+import Chisel._
+import chisel3.Record
+import scala.collection.immutable.ListMap
+
+final case class RecordListMap[T <: Data](eltMap: ListMap[String, T])
```

Comment:
A bit of a nit, but in anticipation of [`SeqMap` in Scala 2.13](https://www.scala-lang.org/api/2.13.2/scala/collection/SeqMap.html) with the faster `VectorMap` concrete implementation (vs. slower `ListMap`), I think it might better to call this `RecordSeqMap`. The type of `elements` will remain `ListMap`, but we could change it to `SeqMap` once we upgrade to 2.13

Reply:
Alternatively this can be `ListMap` and we can introduce a new one in rocket-chip Scala 2.13

Reply:
should I just call it "RecordMap"?

Reply:
RecordMap would be my preference - we can just make it part of the API description that it's ordered

Reply:
I renamed this to RecordMap